iothub_client/src/iothubtransport_amqp_telemetry_messenger.c

```diff
@@ -1024,7 +1024,10 @@ static void invoke_callback(const void* item, const void* action_context, bool*
 #pragma warning(push)
 #pragma warning(disable:4305) // Allow typecasting to smaller type on 64 bit systems, since we control ultimate caller.
 #endif
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wvoid-pointer-to-enum-cast"
```

Comment:
to align with above `#if defined(_MSC_VER)`. Also to tighten the exposure of this since Mac seems to be the only platform needing the fix right now.
```suggestion
#if defined(__clang__) && defined(__APPLE__)
#pragma clang diagnostic push
#pragma clang diagnostic ignored "-Wvoid-pointer-to-enum-cast"
#endif
```